### DISCUSSION TITLE
Adoins new framework for Deno

### DISCUSSION THREAD

**Discussion by baselrabia** [adonisjs/core#3429] [Ideas]
## Hello Adonis
recently I was watching Ryan Dahl in his talk "10 Things I Regret About Node.js " at js conference 2018 and he was talking about his 10 regrets when he made nodejs and announce the new runtime environment for typescript and javascript building in rust with the module concept for package dependencies and it seems inspiring to give a try, so I searched about frameworks for deno and find that they might need something like adonisjs with its power to join the race of the frameworks, so creating a new framework like adonisjs for deno might be a good idea, what do you think ?

### Comparison with Node.js
Deno and Node.js are both runtimes built on Google's V8 JavaScript engine, the same engine used in Google Chrome. They both have internal event loops and provide command-line interfaces for running scripts and a wide range of system utilities.

### Deno mainly deviates from Node.js in the following aspects:

1. Uses ES Module as the default module system, instead of CommonJS.
2. Uses URLs for loading local or remote dependencies, similar to browsers.
3. Includes a built-in package manager for resource fetching, thus no need for npm.
4. Supports TypeScript out of the box, using a snapshotted TypeScript compiler with caching mechanisms.
5. Aims for better compatibility with browsers with a wide range of Web APIs.
6. Restricts file system and network access by default in order to run sandboxed code.
7. Redesigns the API to utilize promises, ES6 and TypeScript features.
8. Minimizes core API size, while providing a large standard library with no external dependencies.
9. Uses message-passing channels for invoking privileged system APIs and using bindings.

### Some Links you should consider

1. [Deno Wikipedia ](https://en.wikipedia.org/wiki/Deno_(software))
2. [Ryan Dahl in his talk "10 Things I Regret About Node.js " ](https://www.youtube.com/watch?v=M3BM9TB-8yA)

**Comment by RomainLanz** [comment]
Hey @baselrabia! 👋🏻 

This is pure marketing.
Nearly all those points are wrong.

Adonis will not migrate to Deno or other JS runtime.

Closing this and locking it to avoid any heat up.
